### DISCUSSION TITLE
Can I run utils rule from CLI?

### DISCUSSION THREAD

**Discussion by andylokandy** [ast-grep/ast-grep#1484] [Q&A] [answered]
Can I define rules in the utils folder and invoke them from `sg scan`?

The situation is that my project has a tool to invoke sg command and consume its json output. I'd like to call the rule with rule id rather than rule file path (in the utils folder). And I can not put it in the rules folder because those rules are not supposed to generate any warning in daily `sg scan`.

**Comment by HerringtonDarkholme** [comment]
No plan to support it in ast-grep CLI itself. You can write a scripting to map id to file and invoke corresponding rule file.

**Comment by andylokandy** [comment]
There may be another workaround: allowing `sg scan --filter xxx --include-off` to run the rules noted with `severity: off`. Is it appropriate?

**Comment by HerringtonDarkholme** [comment, marked as answer]
There is a TODO option of overriding rule severity https://github.com/ast-grep/ast-grep/issues/1061.

**Reply by HerringtonDarkholme** [reply]
@andylokandy this should be implemented by https://github.com/ast-grep/ast-grep/pull/1561 and https://github.com/ast-grep/ast-grep/pull/1546

**Reply by HerringtonDarkholme** [reply]
released in 0.29.0

**Reply by andylokandy** [reply]
Thank you!
